sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala

```diff
@@ -85,6 +90,43 @@ class HiveExternalCatalogVersionsSuite extends SparkSubmitTestUtils {
     new File(tmpDataDir, name).getCanonicalPath
   }
 
+  private def getFileFromUrl(urlString: String, targetDir: String, filename: String): Boolean = {
+    val conf = new SparkConf
+    val securityManager = new SecurityManager(conf)
+    val hadoopConf = new Configuration
+
+    val outDir = new File(targetDir)
+    if (!outDir.exists()) {
+      outDir.mkdirs()
+    }
+
+    try {
+      val result = Utils.doFetchFile(urlString, outDir, filename, conf, securityManager, hadoopConf)
+      result.exists()
+    } catch {
+      case ex: Exception => logWarning("Could not get file from url " + urlString + ": "
+        + ex.getMessage)
+        false
+    }
+  }
+
+  private def getStringFromUrl(urlString: String, encoding: String = "UTF-8"): String = {
+    val outDir = Files.createTempDirectory("string-")
```

Comment:
Use `Utils.createTempDir`; it handles deleting the directory for you later.
 
Another option is to use `File.createTempFile` + `File.deleteOnExit`. Same result.